tests/boot/with_mcumgr/pytest/test_downgrade_prevention.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import time
+import logging
+
+from pathlib import Path
+from twister_harness import Device, MCUmgr
+from conftest import wait_for_message
+
+
+logger = logging.getLogger(__name__)
+
+
+def test_downgrade_prevention(dut: Device, mcumgr: MCUmgr):
+    """
+    Verify that the application is not downgraded
+    1) Device flashed with MCUboot and an application that contains SMP server.
+       Version of MCUboot is 1.1.1
+    2) Prepare an update of an application containing the SMP server, where
+       MCUboot version is 0.0.0 (lower than version of the original app)
```

Comment:
I do not think you mean "MCUboot version" but rather application version set for MCUboot or something like this. I doubt you are able to update MCUboot.